library/manageiq_custom_attributes.py

```diff
@@ -0,0 +1,254 @@
+
+    def delete_custom_attribute(self, ca, ca_href, entity_type, entity_id):
+        """ Returns the deleted custom attribute
+        """
+        try:
+            url = '{api_url}/{entity_type}/{id}/custom_attributes'.format(
+                    api_url=self.api_url,
+                    entity_type=ManageIQ.supported_entities[entity_type],
+                    id=entity_id)
+            ca_object = {'name': ca['name'], 'href': ca_href}
+            result = self.client.post(url, action='delete', resources=[ca_object])
+            self.changed = True
+            return result['results']
+        except Exception as e:
+            self.module.fail_json(msg="Failed to delete the custom attributes {ca}. Error: {error}".format(ca=ca, error=e))
+
+    def delete_custom_attributes(self, entity_type, entity_name, custom_attributes):
+        """ Deletes the custom attributes from the entity, if exist
+
+        Returns:
+            whether or not a change took and a short message including the
+            deleted custom attributes
+        """
+        deleted = []
+        entity_id = self.find_entity_by_name(entity_type, entity_name)
+        if entity_id:
+            ca_list = eval(custom_attributes)
+            entity_cas = self.get_entity_cas(entity_type, entity_id)
+            for new_ca in ca_list:
+                ca_href = next((ca['href'] for ca in entity_cas if ca['name'] == new_ca['name']), None)
+                if ca_href:
+                    deleted.extend(self.delete_custom_attribute(new_ca, ca_href, entity_type, entity_id))
+            return dict(
+                msg="Successfuly deleted the following custom attributes from {entity_name} {entity_type}: {deleted}".format(
+                    entity_name=entity_name, entity_type=entity_type, deleted=deleted),
+                changed=self.changed
+            )
+        else:
+            self.module.fail_json(msg="Failed to add the custom attributes. {entity_type} {entity_name} does not exist".format(entity_type=entity_type, entity_name=entity_name))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            entity_name=dict(required=True, type='str'),
+            entity_type=dict(required=True, type='str',
+                             choices=['provider', 'vm']),
+            state=dict(default='present',
+                       choices=['present', 'absent']),
+            custom_attributes=dict(required=True, type='str'),
+            miq_url=dict(default=os.environ.get('MIQ_URL', None)),
+            miq_username=dict(default=os.environ.get('MIQ_USERNAME', None)),
+            miq_password=dict(default=os.environ.get('MIQ_PASSWORD', None)),
+        )
+    )
+
+    for arg in ['miq_url', 'miq_username', 'miq_password']:
+        if module.params[arg] in (None, ''):
+            module.fail_json(msg="missing required argument: {}".format(arg))
```

Comment:
I'm pretty sure it won't work cause required and default are mutually exclusive.

Anyway, this appears in all 3 modules so I'll address it in a separate PR.